polygeist/lib/Dialect/Polygeist/Utils/TransformUtils.cpp

```diff
@@ -23,14 +23,21 @@ using namespace mlir;
 // Utilities functions
 //===----------------------------------------------------------------------===//
 
+static constexpr StringLiteral linkageAttrName = "llvm.linkage";
 bool mlir::isLinkonceODR(FunctionOpInterface func) {
-  constexpr StringRef linkageAttrName = "llvm.linkage";
   if (!func->hasAttr(linkageAttrName))
     return false;
   auto attr = cast<LLVM::LinkageAttr>(func->getAttr(linkageAttrName));
   return attr.getLinkage() == LLVM::Linkage::LinkonceODR;
 }
 
+void mlir::privatize(FunctionOpInterface func) {
+  func->setAttr(
```

Comment:
Do we need this function at all ? Can you look the LLVM IR generated if we simply make the function private  and do not set any attribute. I am thinking the MLIR private visibility would lower to LLVM's private linkage and if so no need to have this function at all.